src/view/com/post-thread/PostThreadItem.tsx

```diff
@@ -758,6 +760,84 @@ function ExpandedPostDetails({
   )
 }
 
+function PostTimestamp({post}: {post: AppBskyFeedDefs.PostView}) {
+  const t = useTheme()
+  const {_, i18n} = useLingui()
+  const control = Prompt.usePromptControl()
+
+  const indexedAt = new Date(post.indexedAt)
+  const createdAt = AppBskyFeedPost.isRecord(post.record)
+    ? new Date(post.record.createdAt)
+    : new Date(post.indexedAt)
+
+  // backdated if createdAt is 24 hours or more before indexedAt
+  const isBackdated =
+    indexedAt.getTime() - createdAt.getTime() > 24 * 60 * 60 * 1000
+
+  const orange = t.name === 'light' ? colors.warning.dark : colors.warning.light
+
+  return (
+    <>
+      {isBackdated && (
+        <Button
+          label={_(msg`Archival post`)}
+          accessibilityHint={_(
+            msg`Show information about when this post was created`,
+          )}
+          onPress={e => {
+            e.preventDefault()
+            e.stopPropagation()
+            control.open()
+          }}>
+          {({hovered, pressed}) => (
+            <View
+              style={[
+                a.flex_row,
+                a.align_center,
+                a.rounded_full,
+                t.atoms.bg_contrast_25,
+                (hovered || pressed) && t.atoms.bg_contrast_50,
+                {
+                  gap: 3,
+                  paddingHorizontal: 6,
+                  paddingVertical: 3,
+                },
+              ]}>
+              <CalendarClockIcon fill={orange} size="sm" aria-hidden />
+              <NewText
+                style={[
+                  a.text_xs,
+                  a.font_bold,
+                  a.leading_tight,
+                  t.atoms.text_contrast_medium,
+                ]}>
+                <Trans>Archival</Trans>
+              </NewText>
+            </View>
+          )}
+        </Button>
+      )}
+      <NewText style={[a.text_sm, t.atoms.text_contrast_medium]}>
+        {niceDate(i18n, post.indexedAt)}
+      </NewText>
+
+      <Prompt.Basic
+        control={control}
+        title={_(msg`Archival post`)}
+        description={_(
+          msg`This post claims to have been created at ${niceDate(
```

Comment:
```suggestion
          msg`This post claims to have been created on ${niceDate(
```
wording is great imho, but I think using `on` would flow better and avoid repeating `at`